src/gsy_e/models/strategy/energy_parameters/virtual_heat_pump.py

```diff
@@ -0,0 +1,196 @@
+
+        self._water_supply_temp_C: [DateTime, float] = EnergyProfile(
+            water_supply_temp_C_profile, water_supply_temp_C_profile_uuid,
+            profile_type=InputProfileTypes.IDENTITY)
+
+        self._water_return_temp_C: [DateTime, float] = EnergyProfile(
+            water_return_temp_C_profile, water_return_temp_C_profile_uuid,
+            profile_type=InputProfileTypes.IDENTITY)
+
+        self._dh_water_flow_m3: [DateTime, float] = EnergyProfile(
+            dh_water_flow_m3_profile, dh_water_flow_m3_profile_uuid,
+            profile_type=InputProfileTypes.IDENTITY)
+
+    def serialize(self):
+        """Return dict with the current energy parameter values."""
+        return {
+            **super().serialize(),
+            "water_supply_temp_C": self._water_supply_temp_C.input_profile,
+            "water_supply_temp_C_uuid": self._water_supply_temp_C.input_profile_uuid,
+            "water_return_temp_C": self._water_return_temp_C.input_profile,
+            "water_return_temp_C_uuid": self._water_return_temp_C.input_profile_uuid,
+            "dh_water_flow_m3_profile": self._dh_water_flow_m3.input_profile,
+            "dh_water_flow_m3_profile_uuid": self._dh_water_flow_m3.input_profile_uuid,
+        }
+
+    def _rotate_profiles(self, current_time_slot: Optional[DateTime] = None):
+        self._water_return_temp_C.read_or_rotate_profiles()
+        self._water_supply_temp_C.read_or_rotate_profiles()
+        self._dh_water_flow_m3.read_or_rotate_profiles()
+        self.state.delete_past_state_values(current_time_slot)
+
+    def _calc_energy_to_buy_maximum(self, time_slot: DateTime) -> float:
+        max_energy_consumption = self._storage_temp_to_energy(self._max_temp_C, time_slot)
+        assert max_energy_consumption > -FLOATING_POINT_TOLERANCE
+        return min(self._max_energy_consumption_kWh, max_energy_consumption)
+
+    def _calc_energy_to_buy_minimum(self, time_slot: DateTime) -> float:
+        max_energy_consumption = self._storage_temp_to_energy(
+            self.state.get_storage_temp_C(time_slot), time_slot)
+        assert max_energy_consumption > -FLOATING_POINT_TOLERANCE
+        return min(self._max_energy_consumption_kWh, max_energy_consumption)
+
+    def _calc_temp_decrease_K(self, time_slot: DateTime) -> float:
+        dh_supply_temp = self._water_supply_temp_C.profile[time_slot]
+        dh_return_temp = self._water_return_temp_C.profile[time_slot]
+        m_m3 = self._dh_water_flow_m3.profile[time_slot]
+        m_kg_per_sec = m_m3 * 1000 / 3600
+        q_out = m_kg_per_sec * WATER_SPECIFIC_TEMPERATURE * (dh_supply_temp - dh_return_temp)
+        temp_differential_per_sec = -q_out / (
+                WATER_DENSITY * WATER_SPECIFIC_TEMPERATURE * self._tank_volume_l)
+        temp_decrease = temp_differential_per_sec * GlobalConfig.slot_length.total_seconds()
+        assert temp_decrease <= 0.0
```

Comment:
Thanks for this assert! Maybe something for an enhancement story:
We should somehow shield the user from running into this and validate the input profiles for the fact that the supply_temp is always grater than the return temperature.What do you think?

Reply:
Hmmmm, let me add it even now. It is probably easy to add such a check in the market_cycle method.

Reply:
I do not see this being implemented. If you agree, let me add a story.

Reply:
It is at the very bottom of this file:
https://github.com/gridsingularity/gsy-e/pull/1695/commits/2d3e9941b45ba6eaaab4e350747391cdd857d126#diff-0ed9e7c23f93c5256ee1bb8c050f874ab97a9164d637c73d289bee266fdbe546R216

Reply:
Ah, thanks. What I really meant was to validate the profiles when reading (or uploading in the gsy-web case) in order to not let the simulation start and then get surprised by a failing simulation later. What do you think?

Reply:
Hmmm, this needs to be done in the gsy-web then and I do not think that I have an open PR here. As a matter of fact, this strategy is only supposed to live in gsy-e. So I am not sure whether something needs to be done here

Reply:
It could be beneficial on the gsy-e side, too, IMO Because this would save us some re-runs of the simulation if this condition is not valid for the provided profile files (worst case at the end of the simulation time).
Again, nothing urgent, but IMO very good to have.